Fill.py

```diff
@@ -938,6 +949,9 @@ def failed(warning: str, force: typing.Union[bool, str]) -> None:
                             err.append(f"Cannot place {item_name} into already filled location {location}.")
                     else:
                         err.append(f"Mismatch between {item_name} and {location}, only one is an event.")
+                else:  # Execute if break statement wasn't hit: Placement was unsuccessful, put item back if from_pool
+                    if from_pool:
+                        world.itempool.append(item)
```

Comment:
insert maybe?

Reply:
Like, back at the original index? Can I hear the argument?

Reply:
Oh no at this point I like the new solution way better lol

Is there anything non-deterministic in this code? If not, this would just make it different, but still deterministic, right?

Reply:
correct, which could possibly have negative consequences. i'm not aware of any off the top of my head, but it is a concern

Reply:
I guess maybe someone might rely on the idea that all of their items are next to each other in the multiworld itempool, because they added them all at the same time? Idk

Reply:
Yeah I think I'm just going back to the way it was before treble suggested to pop it right away